src/theory/arith/nl/equality_substitution.cpp

```diff
@@ -22,6 +22,22 @@ namespace theory {
 namespace arith {
 namespace nl {
 
+namespace {
+struct ShouldTraverse : public SubstitutionMap::ShouldTraverseCallback
```

Comment:
FYI, for the substitute utitlity I had used, it was also important not to traverse beneath certain arithmetic operators, see comment here: https://github.com/cvc5/cvc5/blob/master/src/theory/arith/arith_utilities.cpp#L238

This might lead to issues for transcendentals.  I think I'd recommend changing this method to only return true for "core" arithmetic operators.

Reply:
Ah, right, I somehow forgot about this detail. I added this.